capstone/src/main/java/com/google/sps/Objects/Post.java

```diff
@@ -49,6 +49,21 @@ public static Post getPostEntity(Entity entity) {
     return new Post(postId, authorId, postText, comments, challengeName, timestamp, img, likes);
   }
 
+  @Override 
+  public boolean equals(Object other) {
+    if (other == null) return false;
+    if (other == this) return true;
+    if (!(other instanceof Post)) return false;
+    Post post = (Post) other;
+    return timestamp == post.timestamp &&
+      authorId.equals(post.authorId) &&
+      postText.equals(post.postText) &&
+      challengeName.equals(post.challengeName) &&
+      img.equals(post.img) &&
+      likes.containsAll(post.likes) &&
+      comments.containsAll(post.comments);
+  }
+
```

Comment:
I think Andrew suggested that instead of doing this you can compare the json for testing purposes. This is so that if the class is updated, you don't have to then also make sure this equals method is updated

Reply:
Yeah I think using json string comparison is a valid way to check comparison without having to update this function every time you add a field. You may have to make a temporary object and modify it to make sure you're only comparing fields which are REQUIRED for equality (i.e. I think you had cases where a property could be missing, could be null, or could be an empty array and then you'd have to determine which of those cases SHOULD be considered equal).